Bottom: 4b6adce26ba8f646a5c258134794671aa8803aba
Top:    9c4b4f44aa55a7dce69459ea945cbc67ee125aff
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-09-30 23:13:27 +0100

Fix text bounds to include the area cleared into a polygon

Fixes re-clearing the polygon when a closely touching object is
incrementally updated.


---

diff --git a/src/misc.c b/src/misc.c
index 2153a1a..df34363 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -465,6 +465,14 @@ SetTextBoundingBox (FontTypePtr FontPtr, TextTypePtr Text)
       RotateBoxLowLevel (&Text->BoundingBox,
                          Text->X, Text->Y, Text->Direction);
     }
+
+  /* the bounding box covers the extent of influence
+   * so it must include the clearance values too
+   */
+  Text->BoundingBox.X1 -= PCB->Bloat;
+  Text->BoundingBox.Y1 -= PCB->Bloat;
+  Text->BoundingBox.X2 += PCB->Bloat;
+  Text->BoundingBox.Y2 += PCB->Bloat;
 }
 
 /* ---------------------------------------------------------------------------
diff --git a/src/polygon.c b/src/polygon.c
index 50912b2..70adef3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -905,7 +905,7 @@ UnsubtractText (TextType * text, LayerType * l, PolygonType * p)
 
   if (!TEST_FLAG (CLEARLINEFLAG, text))
     return 0;
-  if (!(np = RoundRect (b->X1, b->X2, b->Y1, b->Y2, PCB->Bloat + 100)))
+  if (!(np = RoundRect (b->X1, b->X2, b->Y1, b->Y2, 100)))
     return -1;
   if (!Unsubtract (np, p))
     return 0;
